packages/core/lib/src/core_html_widget.dart

```diff
@@ -176,13 +175,13 @@ class _HtmlWidgetState extends State<HtmlWidget> {
     if (_future != null) {
       return FutureBuilder<Widget>(
         builder: widget.buildAsyncBuilder ?? _buildAsyncBuilder,
-        future: _future.then(_tshWidget),
+        future: _future!.then(_tshWidget),
```

Comment:
Is `!` required? I thought its type is inferred automatically by the surrounding `if` statement?

Reply:
This is _really_ annoying, but dart can't do that.
The problem is that `_future` is a field, not a local variable. Fields in dart are always implicitly a getter/setter, and subclasses could _theoretically_ override the getter to e.g. randomly return null and non-null values, so it can't be trusted to return null the second time we check it. [The dart team is discussing possible fixes](https://github.com/dart-lang/language/issues?q=is%3Aissue+is%3Aopen+promotion+label%3Afield-promotion), but for now the only workaround is to introduce a temporary local variable. In this short case I don't hink that is worth it.